sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/SuppressReviveManager.java

```diff
@@ -92,161 +63,60 @@ public void run() {
-                case INITIAL:
-                    logger.error("Invalid state transition.  End state should never be INITIAL");
-                    return;
-                case WAITING_FOR_OFFER:
-                    switch (current) {
-                        case REVIVED:
-                            logger.debug("Already revived, no need to revive again and wait for offers.");
-                            target = State.REVIVED;
-                            break;
-                        default:
-                            revive();
-                    }
-                    break;
-                case REVIVED:
-                    switch (current) {
-                        case WAITING_FOR_OFFER:
-                            // The only acceptable transition
-                            break;
-                        default:
-                            logTransitionError(current, target);
-                            return;
-                    }
-                    break;
-                case SUPPRESSED:
-                    switch (current) {
-                        case REVIVED:
-                            suppress();
-                            break;
-                        case WAITING_FOR_OFFER:
-                            logTransitionWarning(current, target);
-                            return;
-                        default:
-                            logTransitionError(current, target);
-                            return;
-                    }
-                    break;
-            }
-
-            if (state.compareAndSet(current, target)) {
-                logger.debug("Transitioned from '{}' to '{}'", current, target);
-            } else {
-                logger.error("Failed to transitioned from '{}' to '{}'", current, target);
-                return;
-            }
+    private void suppressOrRevive() {
+        Set<PodInstanceRequirement> newCandidates = getRequirements(planCoordinator.getCandidates());
+        if (newCandidates.isEmpty()) {
+            logger.debug("No candidates found");
+            suppress();
+            candidates = newCandidates;
+            return;
         }
-    }
 
-    private void logTransitionError(State start, State end) {
-        logger.error("Invalid transition from '{}' to '{}'.", start, end);
-    }
+        newCandidates.removeAll(candidates);
 
-    private void logTransitionWarning(State start, State end) {
-        logger.warn("Unexpected transition from '{}' to '{}'.", start, end);
-    }
+        logger.debug("Old candidates: {}", candidates);
+        logger.debug("New candidates: {}", newCandidates);
 
-    private void suppressOrRevive() {
-        boolean hasOperations = planManagers.stream()
-                .anyMatch(planManager -> PlanUtils.hasOperations(planManager.getPlan()));
-        if (hasOperations || hasTasksNeedingRecovery()) {
-            transitionState(State.WAITING_FOR_OFFER);
+        if (newCandidates.isEmpty()) {
```

Comment:
IMO the comments attached to this PR as a description would make sense here as documentation.